### PR TITLE
chore(@desktop/onboarding): splash screen removed and used as app loading animation

### LINKED ISSUE #5753: Display the Splash screen while the app is being loading
Currently we're displaying the Splash screen when one starts the app (before displaying login/onboarding screen). The point is that in that moment we don't have some heavy actions to do on the backend and we actually just delays the login screen waiting for the end of animation on the Splash screen (since everything is prepared much faster than the animation length).

What we need to do:
- Display Splash screen once login process is about to start (after user entered correct credentials and pressed enter)
- Once Splash screen is displayed animation should be run in a loop for all the time app loading lasts
- Remove Splash screen immediately when complete UI and all backend data are loaded/prepared (we don't need to wait for the animation to be done, so just stop it whenever loading on the backend is done)
- Display the app, which must be fully operable from that moment

### PR DESCRIPTION
**Pull request by saledjenic**
fixes #5753

This is created just as a demo, we can either use splash screen as loading app animation or keep it on the app start and add new animation for loading app. But all in all when user runs the app it's ready immediately so splash screen on the start just delays the app start.

https://user-images.githubusercontent.com/86303051/180407686-3c2a6a87-2667-45a5-9dce-34e43c029309.mov

### CODE REVIEW

**Review by osmaczko — APPROVED** [review]
Looks good, logically it makes more sense to have a splash screen as an app loading animation :ok_hand:

File: ui/app/AppLayouts/Onboarding/OnboardingLayout.qml [hunk 1]
```diff
@@ -66,6 +67,10 @@ OnboardingBasePage {
             {
                 return loginViewComponent
             }
+            if (root.startupStore.currentStartupState.stateType === Constants.startupState.loadingAppAnimation)
```
[the remark this answers is not in the document]
Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/OnboardingLayout.qml:70]: @igor-sirotin is this just a comment or you're suggesting some changes? 
If you have code like:
```
if (cond1) { // something...}
if (cond2) { // something...}
.
.
.
if (condN) { // something...}
```
there is no difference from this:
```
if (cond1) { // something...}
else if (cond2) { // something...}
.
.
.
else if (condN) { // something...}
```

**Review by stefandunca — COMMENTED** [review]
@saledjenic I can't tell how the splash screen is displayed while the business logic and presentation layer is loading. It looks to me that after all the processing, the splash screen animation is shown without being interrupted. Maybe I miss seeing the events leading to the asynchronous splash screen loading state?

File: ui/app/AppLayouts/Onboarding/OnboardingLayout.qml (hunk 1 above)
[the remark this answers is not in the document]
Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/OnboardingLayout.qml:70]: @stefandunca please check example above and let me know is there any difference.

[the remark this answers is not in the document]
Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/OnboardingLayout.qml:70]: Also want to update this and call setup account in an async way. That will be an improvement within this PR.

[the remark this answers is not in the document]
Comment (stefandunca) [reply ui/app/AppLayouts/Onboarding/OnboardingLayout.qml:70]: > @stefandunca please check example above and let me know is there any difference. (I updated the comment, actually that was what I meant, but didn't type it like that initially).

@saledjenic  sorry, I replied to the notification email and didn't check the update :). Thanks for clarifications

**Comment by stefandunca** [comment]
> On my machine there is no significant difference between these 2 approaches, but that's the reason why I splitted them into 2 independent commits so we can discuss/decide whether to merge both or exclude the second one.

If we don't have a clear improvement, merging a more complex and error-prone implementation is not worth the trouble.

If the async after onboarding processing is not noticeable, I'm wondering what is causing the hanging of the UI.
It looks to me that showing the splash screen is worth only for the duration of the UI freeze and stopping it with a rapid fade when the UI is ready for interaction.

**Comment by glitchminer** [comment]
### **Issues Found**

On slower Windows 11 laptop the splash screen no longer displays at the expected size with the logo.

https://user-images.githubusercontent.com/50769329/181235143-15d5e005-65ce-4d3b-a83c-30e1336efb76.mp4

Observation only: On M1 Max, the splash screen no longer appears at all.

**Test info**
Build: StatusIm-Desktop-220725-104548-f2775d-pr6579.exe
OS: Win11

**Comment by glitchminer** [comment]
Added issue 2. to issues found:  https://github.com/status-im/status-desktop/pull/6579#issuecomment-1196604147

**Comment by saledjenic** [comment]
@glitchminer what you've found was not noticeable for me, it might be because of machine speed, but I think that I know what's the problem. Will sort it out. Thanks for the test. 
I am now into some other ticket and this is not that urgent thing, that's why I set `don't merge yet` label, will handle that these days and ask you to check again then.

File: ui/app/AppLayouts/Onboarding/OnboardingLayout.qml (hunk 1 above)
[the remark this answers is not in the document]
Comment (osmaczko) [reply ui/app/AppLayouts/Onboarding/OnboardingLayout.qml]: May be related to the discussion: https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html. It describes well why else after return decreases readability.

**Comment by saledjenic** [comment]
@all closing this PR in favour of https://github.com/status-im/status-desktop/pull/6755

**Pull request closed without merging**